apps/web/app/modules/Courses/Lesson/LessonItems/Question/Question.tsx

```diff
@@ -66,15 +72,10 @@ export const Question = ({ isSubmitted, content, questionsArray }: QuestionProps
   }, [isQuiz, isSubmitted]);
 
   const handleClick = async (id: string) => {
-    await markLessonItemAsCompleted({
-      lessonItemId: questionId,
-      lessonId,
-      courseId,
-    });
-
     if (isSingleQuestion) {
       setSelectedOption([id]);
       await sendAnswer([id]);
+      !isCompleted && !isQuiz && updateLessonItemCompletion(lessonItemId);
```

Comment:
we are using duplicate code here, it would be a good idea to extract the duplicate code into a separate function and use it here. This will make it easier for us to make changes in the future. 
`const handleCompletion = () => {
  if (!isCompleted && !isQuiz) {
    updateLessonItemCompletion(lessonItemId);
  }
};
`